GameServer/Logic/Maps/SummonersRift.cs

```diff
@@ -349,37 +419,91 @@ public override bool Spawn()
                 MinionSpawnPosition.SPAWN_RED_BOT,
                 MinionSpawnPosition.SPAWN_RED_MID,
             };
-
-            if (_waveNumber < 3)
+            foreach (var pos in positions)
             {
-                for (var i = 0; i < positions.Count; ++i)
+                uint inhibitorId = 0;
+                var waypoints = new List<Vector2>();
+                switch (pos)
```

Comment:
Dictionaries strike again!

```CS

var _spawnToWaypoints = new Dictionary<MinionSpawnPosition, Tuple<List<Vector2>, uint>>
{
    {MinionSpawnPosition.SPAWN_BLUE_BOT, new Tuple<List<Vector2>, uint>(_blueBotWaypoints, 0xff26ac0f)},
    {MinionSpawnPosition.SPAWN_BLUE_MID, new Tuple<List<Vector2>, uint>(_blueMidWaypoints, 0xffff8f1f)},
    {MinionSpawnPosition.SPAWN_BLUE_TOP, new Tuple<List<Vector2>, uint>(_blueTopWaypoints, 0xff6793d0)},
    {MinionSpawnPosition.SPAWN_RED_BOT, new Tuple<List<Vector2>, uint>(_redBotWaypoints, 0xff9303e1)},
    {MinionSpawnPosition.SPAWN_RED_MID, new Tuple<List<Vector2>, uint>(_redMidWaypoints, 0xff4a20f1)},
    {MinionSpawnPosition.SPAWN_RED_TOP, new Tuple<List<Vector2>, uint>(_redTopWaypoints, 0xffd23c3e)}
}
waypoints = _spawnToWaypoints[pos].Item1;
inhibitorId = _spawnToWaypoints[pos].Item2;
```
https://msdn.microsoft.com/en-us/library/system.tuple.aspx

The dict could also be defined as a constant